config/stored_requests.go

```diff
@@ -316,6 +314,8 @@ func makeIdList(numSoFar int, numArgs int) string {
 }
 
 type InMemoryCache struct {
+	// Identify the type of memory cache. "none", "unbounded", "lru"
+	Type string `mapstructure:"type"`
```

Comment:
I think this should be an enum? To identify the 3 legal values.

Reply:
Yes, would be more proper with an enum. :)

Reply:
Actually, on second thought I think an enum would add overhead without any benefit. The config value only gets set from Viper, so any incoming string will be forced into the `MemCacheType` type, eliminating any constraints on the value that go "enum" could enforce. The code never changes this later, which is where type safety comes in. And finally, we have the `validate()` method we call after reading in the configuration that will enforce the value to be one of the list.